 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>
<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$

<header style="background-color:#e7e2ee; color:#461d7c; line-height:2em; padding-left:1em; padding-right:1em; bottom:0; float:bottom; text-align:right; float:right; font-family:serif; font-size:small;">Problem 1.038 ⋮ Moran, Shapiro, Boettner, Bailey, <i>Fundamentals of Engineering Thermodynamics</i>, 9th edition.</header>  

# Problem 1.038   

As shown in the figure, an inclined manometer is used to measure the pressure of the gas within the reservoir. The manometer fluid is mercury, which has a density of 845 lb/ft³. The manometer fluid rises a distance
7 inches along the manometer tube, which is inclined 30º from the horizontal. The atmospheric pressure is 14.7 lbf/in² and the acceleration of gravity is 32.2 ft/s². Determine the gas pressure, in lbf/in². Express the pressure as a gage or a vacuum pressure, as appropriate, in lbf/in².

![ms9⋮1.038](./figures/ms9⋮1-038.png)

In [1]:
# a little Julia housekeeping
using Unitful;       # use unitful.jl units package for Julia

## Given   

Gas reservoir with manometer as described.

In [2]:
# given
       ρ_Hg = 845u"lb/ft^3"  # some people are anal about lining up equations
          g = 32.2u"ft/s^2"
θ_manometer = 30u"°"
L_manometer = (7//12)u"ft"  # julia has a Rational data type that treats fractions exactly

7//12 ft

## Find  

$P_{gas}\ [\small \ce{psig}]$

## Assumptions  

Elevation increases in the upward direction.

## Get Organized

This is one of the few scenarios where working in gage pressure makes sense. 
We are tasked with determining the pressure difference across the gauge (our manometer). 
As always with hydrostatic pressure scenarios, we start with either a pressure we know or the pressure we seek, then traverse the manometer from that one to the other. 
We'll solve this one from both perspectives.  

Remember, hydrostatic pressure is a function of **vertical** depth/elevation, so we must resolve the vertical component of the manometer length.

#### Solving with gage pressure   

Our two endpoints are the pressure we seek $(P_{gas})$ and the mercury/atmosphere interface, where we know $P_{gage} = 0.$  

Starting at $P_{gas},$

$$
\begin{array}{lcc}
  P_{gas,gage} &+& \Delta P_\ce{Hg} &= 0 \\
  P_{gas,gage} &+& (\rho_\ce{Hg})(-|g|)(-L\sin\theta) &= 0
\end{array}  \\[18pt]
$$  

$$P_{gas,gage} = - \rho_\ce{Hg}|g|L\sin\theta $$  


$$P_{gas,vacuum} = \rho_\ce{Hg}|g|L\sin\theta \tag{1} \label{gage}$$  

#### Solving with absolute pressure   

This time, our two endpoints are the pressure we seek $(P_{gas})$ and the atmosphere, where we were given $P_{atm} = 14.7\,\ce{psia}.$  

Starting at $P_{gas},$

$$
\begin{array}{lcc}
  P_{gas,abs} &+& \Delta P_\ce{Hg} &= P_{atm} \\
  P_{gas,abs} &+& (\rho_\ce{Hg})(-|g|)(-L\sin\theta) &= P_{atm} \\
\end{array}  \\[18pt]
$$   

$$
\begin{array}{ccc}
  P_{gas,abs} &=& P_{atm} - \rho_\ce{Hg}|g|L\sin\theta \\
  P_{gas,gage} + \cancel{P_{atm}} &=& \cancel{P_{atm}} - \rho_\ce{Hg}|g|L\sin\theta 
\end{array}  \\[18pt]
$$  

\begin{array}{rr}
P_{gas,gage} &= - \rho_\ce{Hg}|g|L\sin\theta     \\
P_{gas,vacuum} &= \phantom{-} \rho_\ce{Hg}|g|L\sin\theta \tag{2} \label{abs}
\end{array}  

That's the same result we obtained in Eq.1 earlier.

## Attack   

We're organized and calm and ready to find everything we're supposed to find.  

In [3]:
P_gas_vacuum = ρ_Hg * g * L_manometer * sin(θ_manometer) |> u"lbf/inch^2"

println("P_gas_vacuum = ",P_gas_vacuum)

P_gas_vacuum = 1.7128967050136623 lbf inch^-2


$$ 
\bbox[#dff0d8, 0.5em, border:2.5pt solid #4cae4c]{
\color{#3c763d}
P_\text{gas,vacuum} = 1.713\, \small \ce{psig}
}
$$

<div class="btn-info">&nbsp;<div class="badge"></div><i class="fa fa-code"></i><strong> Julia Notes</strong></div>
<div class="panel-info"><div class="panel-heading">   

1. Julia allows unicode characters in variable names, eg: `𝛀𝚿𝚽 = 1911`, or `Δ𝑓ℎ̅°_Al₂O₃ᶜᴿ = -1675.700kJ/mol`.
1. Julia has a built-in constant `π`, also known as `pi`. Julia, like all computer languages, performs calculations and stores values at various levels of precision. Higher precision means slower computation and more memory consumption. Life is a series of compromises.
</div></div>

In [4]:
# fun with calculation precision

# julia built-in constant
println(π) 
println()

# popular simple trigonometric method to obtain a value for π in
# languages lacking a built-in π constant
println("acos(-1.0) = ",acos(-1.0))
println()

# compare
println("π - acos(-1.0) = ",π - acos(-1.0))
println()

# compare another way
println("π == acos(-1)? ",π == acos(-1))

π = 3.1415926535897...

acos(-1.0) = 3.141592653589793

π - acos(-1.0) = 0.0

π == acos(-1)? false


<div class="panel-info"><div class="panel-heading">   
    
3. The difference doesn't matter for us now, but it is a huge factor in equality comparisons, and it could become a factor in an application with repetitive calculations. All numeric calculations inherently have these slight quirky errors.
</div></div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>